### ISSUE TITLE
Unable to use on port as transition condition in the Behavior Annex

### ISSUE THREAD

**Issue by smithdtyler** [osate/osate2#2147]
Via @Etienne13 

> Input ports may be used in transition of such components, in particular in order to represent mode transitions. In this case, the syntax allows to define the transition condition as follows:
> 
> s0 -[on input]-> s1;

> It seem the current version of the BA plugin does not support this syntax. You can fill in an issue on github and I will have a look at the problem.

## Steps to Reproduce
See example model below

```aadl
 thread group threadA

    features

      input: in event data port message_pm;

  end threadA;


  thread group implementation threadA.impl

    subcomponents

      t0 : thread;

  

    annex behavior_specification {**

    states

    s0: initial final state;

    s1: state;

    transitions

    s0 -[ on input]-> s1;

    **};

  end threadA.impl;
```

## Environment

* **OSATE Version**: 2.6.1
* **Operating System**: Windows 10

**Comment by Etienne13** [comment]
This issue should be fixed after integrating PR #2152.

**Comment by Etienne13** [comment]
@smithdtyler to be consistent with the BA standard:
- s0 should be the identifier of a mode in threadA.impl
- s0 should be a complete state

here is the result:

```aadl
  thread group implementation threadA.impl
    subcomponents
      t0 : thread;
    modes
      s0: initial mode;
    annex behavior_specification {**
    states
      s0: initial complete final state;
      s1: state;
    transitions
      s0 -[ on input]-> s1;
    **};
  end threadA.impl;
```
